### PR TITLE
Implement step definition for login feature

### PR DESCRIPTION
**Pull request by gurunglaxmi891**
This PR adds step definitions for login feature and implements Page Object Model for better test structure and maintainability.

### CODE REVIEW

**Comment by prashant-gurung899** [comment]
Update your description please

File: tests/pageObjects/LoginPage.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class LoginPage {
+    constructor() {
+        this.loginBtnLocator = ".btn.btn-primary.btn-block"; 
+        this.emailFieldLocator = "#spree_user_email";
+        this.passwordFieldLocator = "#spree_user_password";
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js]: Use the term `selector`
```suggestion
        this.passwordFieldSelector = "#spree_user_password";
```

File: tests/pageObjects/LoginPage.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        this.adminDashboardSelector = "div.alert-success";
+        this.errorMessageLocator = "div.alert-danger";
+    }
+
+    async navigate() {
```
Comment (prashant-gurung899) [inline tests/pageObjects/LoginPage.js]:
```suggestion
    async navigateToAdminLoginPage() {
```

**Comment by koebel** [comment]
Does this PR relates to the issue about E2E tests (https://github.com/spree/spree/issues/12235)? Linking to issues helps other developers who are considering to make open source contributions see that someone is already working on this task and avoid wasting time and effort by having multiple people do the same task  🙈

And if there is any dependency of this Draft PR with another open PR towards the original project (maybe with https://github.com/spree/spree/pull/12267), it's also helpful to link them, in particular if you are considering to add your tests as an open source contribution towards the original project at some point in time in the future.

**Pull request merged**